### ISSUE TITLE
Publish DITA topic or map in the context of a DITA root Map for resolving key and conkeyrefs

### ISSUE THREAD

**Issue by raducoravu** [dita-ot/dita-ot#1697]
Some of our users want to preview how a edited DITA topic would look like in XHTML. 
The problem is that the topic might use conkeyrefs so it cannot be published on its own, it somehow would need to be published by specifying also as a context for resolving keys the root map.
The behavior should be similar to having a root map which references only this topic and to publish it.

**Comment by raducoravu** [comment]
:+1: Another Oxygen + DITA OT end user wanted this.

**Comment by raducoravu** [comment]
👍 Another Oxygen user asked for this:

> We are converting all our images to keys and creating maps of keys. We imbed those maps in maps containing our topics. When we run a transformation scenario against those maps, it works perfectly. But when we run a transformation scenario for a single topic, the key is not found and the image does not appear in the output.

**Comment by xephon2** [comment]
+1

**Comment by jelovirt** [comment]
Making this work with old preprocess might be a bit complex, but with map-first processing it should be a bit easier. It might be possible to make this work by creating a wrapper map

```xml
<map>
  <mapref href="<map>" processing-role="resource-only"/>
  <topicref href="<topic>"/>
</map>
```

and then at the end of the preprocessing step remove all files except the topic (and linked images).

**Comment by raducoravu** [comment]
Actually it would be great if this could work also if you publish a submap, you would point a parameter to the master map. Ideally the master map would be used just to extract the key mappings. 
About this remark:

> and then at the end of the preprocessing step remove all files except the topic (and linked images)

why remove them? They should be referenced as resource only anyway...

**Comment by jelovirt** [comment]
@raducoravu I'm not 100% sure the idea that using the wrapper map will work as expected, so that's why I mentioned the idea of removing the other files. But, if we want to publish a submap, then we can't remove other files. Maybe the dynamically created map would work; a similar approach is already used by the list of filters feature, where a filter is created dynamically.

**Comment by raducoravu** [comment]
👍 We had another end user request for publishing a submap in the context of a main DITA Map defining keys for each image

**Comment by raducoravu** [comment]
👍 Another one of our end users:

      I am using keys in my DITA topics. When I apply the transformation scenario for PDF for a DITA map, the keys are being resolved and everything looks fine. When I apply the transformation scenario for PDF for a single DITA topic, the keys are not being resolved and do not show in the PDF.
      Is it possible to use keys and still be able to generate a PDF output for single topics?

**Comment by raducoravu** [comment]
And one more:
https://www.oxygenxml.com/forum/viewtopic.php?f=20&t=16753&p=48016#p48016

**Comment by raducoravu** [comment]
👍 Requested also here:

> From this bookmap, I want to generate two variants of output, but apparently, it is not possible to create two different transformation scenarios in which each of them has a reference to a different rootmap (or is this a very bad idea?). So in this case, the conkeyrefs are resolved with the values of the first warehouse ditamap in the bookmap.

**Comment by raducoravu** [comment]
👍 Requested also here:

https://www.oxygenxml.com/forum/topic18029.html

**Comment by raducoravu** [comment]
👍 Feedback from an Oxygen XML Editor end user:

    Sometimes writers want to publish part of a manual or just one topic.
    While writing and when the “Root map” is set correctly, technical writers see resolved keys.
    However when they publish that one topic, keys/conkeyrefs are empty.

**Comment by raducoravu** [comment]
👍  Feedback from another Oxygen XML Editor end user:

      I often send out single topics for review in PDF. 
      Unfortunately, I cannot create a PDF from just the topic because then the keys are not resolved.  
      As a workaround, I open the map (that has the key definitions), set all other topics in the map to invalid via profiling, and then publish the map.
      It would be handy to publish from Oxygen the topic and all keys are resolved from the map the topic belongs. 
      This key resolving also works in combination DITA Maps Manager > editor: if I have a map open and open a referenced topic, the keys are resolved. Bingo! Could this information not be passed to the OT?

**Comment by raducoravu** [comment]
@infotexture is this issue perhaps fixed by #3412 ?

**Comment by infotexture** [comment]
Indeed.
